flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -154,14 +179,28 @@ private void monitorAndForwardSplits() {
       ScanContext newScanContext;
       if (lastSnapshotId == INIT_LAST_SNAPSHOT_ID) {
         newScanContext = scanContext.copyWithSnapshotId(snapshotId);
+        LOG.debug("Start generating splits for {}", snapshotId);
       } else {
+        if (scanContext.maxSnapshotCountPerMonitorInterval() ==
```

Comment:
Related to an earlier comment. if the default value is `Integer.MAX_VALUE`, we won't need the if-else

Reply:
Updated.